src/main/java/picard/util/StringDistanceUtils.java

```diff
@@ -0,0 +1,253 @@
+
+        for (int j = 0; j < barcodeBytes.length; j++) {
+            numMismatches += freeDistanceForBarcodes(barcodeBytes[j], readSubsequence[j], qualities == null ? null : qualities[j], minimumBaseQuality);
+        }
+
+        return numMismatches;
+    }
+
+    public static int hammingDistance(final byte[] barcodeBases, final byte[] readBases, final byte[] readQualities, final int minimumBaseQuality) {
+        int numMismatches = 0;
+        for (int i = 0; i < barcodeBases.length && readBases.length > i; ++i) {
+            if (SequenceUtil.isNoCall(readBases[i])) {
+                continue;
+            }
+            if (!SequenceUtil.basesEqual(barcodeBases[i], readBases[i])) {
+                ++numMismatches;
+                continue;
+            }
+            if (readQualities != null && readQualities[i] < minimumBaseQuality) {
+                ++numMismatches;
+            }
+        }
+        return numMismatches;
+    }
+
+    static boolean anySmaller(final byte[] values, final int minValue) {
+        for (int i = 0; i < values.length; i++) {
+            if (values[i] < minValue) return true;
+        }
+        return false;
+    }
+
+    static byte[] maskSmaller(final byte[] bases, final byte[] qualities, final int minValue) {
+        final byte[] maskedBases = Arrays.copyOf(bases, bases.length);
+
+        for (int i = 0; i < qualities.length; i++) {
+            if (qualities[i] < minValue) maskedBases[i] = '.';
+        }
+        return maskedBases;
+
+    }
+
+    private static byte[] getBytesMasked(final byte[] readBases, final byte[] readQualities, final int minimumBaseQuality) {
+        final byte[] maskedReadBases;
+        if (readQualities == null || !anySmaller(readQualities, minimumBaseQuality)) {
+            maskedReadBases = readBases;
+        } else {
+            maskedReadBases = maskSmaller(readBases, readQualities, minimumBaseQuality);
+        }
+        return maskedReadBases;
+    }
+
+
+    /**
+     * The FREE (  Filled/truncated Right End Edit ) distance is apt for comparing barcodes when the number of bases
+     * Is predetermined by the sequencing experiment.
+     * @param barcodeBases The bases that the (comparison) barcode has
+     * @param readBases The bases that the read has
+     * @param readQualities the Read qualities
+     * @param minimumBaseQuality The minimum base quality in the read that will be considered informative.
+     * @return the FREE distance (allowing for non-informative bases) between the read and the barcode.
+     * This uses a threshold of 10 meaning that the distances obtained will never exceed 10.
+     */
+    public static int freeDistanceForBarcodes(final byte[] barcodeBases, final byte[] readBases, final byte[] readQualities, final int minimumBaseQuality) {
+        final byte[] maskedReadBases = getBytesMasked(readBases, readQualities, minimumBaseQuality);
+
+        return freeDistance(barcodeBases, maskedReadBases, 10);
```

Comment:
Why the hardcoded threshold of 10?  Would be nice to have as a parameter of the method.